app/src/main/java/com/github/se/travelpouch/model/notifications/NotificationRepository.kt

```diff
@@ -0,0 +1,79 @@
+package com.github.se.travelpouch.model.notifications
+
+import android.util.Log
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.Query
+
+/**
+ * Repository class for managing notifications in the Firestore database.
+ *
+ * @property firestore The Firestore instance used for database operations.
+ */
+class NotificationRepository(private val firestore: FirebaseFirestore) {
+
+  // Reference to the "notifications" collection in Firestore
+  private val notificationCollection = firestore.collection("notifications")
+
+  fun getNewUid(): String {
+    return notificationCollection.document().id
+  }
+
+  /**
+   * Adds a new notification to the Firestore database.
+   *
+   * @param notification The notification to be added.
+   */
+  fun addNotification(notification: Notification) {
+    notificationCollection
+        .document(notification.notificationUid)
+        .set(notification)
```

Comment:
Using .toMap() is not necessary because your NotificationContainer2 class is properly structured and Firestore can serialize it directly. Firestore can handle Kotlin data classes and their properties without needing to convert them to a map.  If you look Firebase, you can see examples of notifications !